tensorflow/core/util/use_cudnn.cc

```diff
@@ -38,12 +38,15 @@ namespace tensorflow {
 
 bool CudnnUseFrontend() {
 #if GOOGLE_CUDA && CUDNN_VERSION >= 8100
-  bool value = false;
-  Status status = ReadBoolFromEnvVar("TF_CUDNN_USE_FRONTEND", false, &value);
-  if (!status.ok()) {
-    LOG(ERROR) << status;
-  }
-  return value;
+  static bool result = [] {
+    bool value = false;
+    Status status = ReadBoolFromEnvVar("TF_CUDNN_USE_FRONTEND", false, &value);
+    if (!status.ok()) {
+      LOG(ERROR) << status;
+    }
+    return value;
+  }();
+  return result;
```

Comment:
Better to move the #if logics inside the lambda function

Reply:
Yes, done.